### PR TITLE
Fix Linux install folder permissions

### PR DESCRIPTION
**Pull request by timtmok**
### Intent
Address #13966

### Approach
Add directory permissions for the top-level directory. I didn't look too deeply into why the electron-forge/webpack upgrade caused this as a quick check didn't reveal anything obvious.

The group and world have execute set because it's required for listing directory contents. Other programs are installed this way and it's what we used to have before the regression.

### Automated Tests
n/a

### QA Notes
The tarballs were also affected but it wouldn't have been noticed since the permissions were for owner only and the user would be the owner in that case. This has been fixed anyway as the directory permission is set prior to packaging the installer and tarball.

### Documentation
n/a

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Review by kevinushey — APPROVED** [review]
LGTM!

**Comment by kevinushey** [comment]
https://github.com/rstudio/rstudio/pull/13979/files#diff-78657f965037cf1abab0e9c739a1079f9239e572d0389e497495a2ece5f643dfR219

Is this still needed?

File: src/node/desktop/CMakeLists.txt [hunk 1]
```diff
@@ -211,6 +211,10 @@ elseif(LINUX)
 
    install(
       DIRECTORY "${ELECTRON_BUILD_DIR}/out/RStudio-linux-${ELECTRON_ARCH}/"
+      DIRECTORY_PERMISSIONS
+      OWNER_READ OWNER_WRITE OWNER_EXECUTE
+      GROUP_READ GROUP_EXECUTE
+      WORLD_READ WORLD_EXECUTE
       DESTINATION "${RSTUDIO_INSTALL_ELECTRON}"
       USE_SOURCE_PERMISSIONS
```
Comment (kevinushey) [inline src/node/desktop/CMakeLists.txt:219]: Is this line still needed? I wonder if the issue reflects an issue with the permissions on the directory created here now that webpack etc. has been updated?

Comment (timtmok) [reply src/node/desktop/CMakeLists.txt:219]: I'm not totally familiar with `cmake` so I didn't try without the `USE_SOURCE_PERMISSIONS` option. It seemed that only the top-level directory was affected by the change to be user only accessible.

My thought is that it is needed to preserve the file permissions.

Comment (kevinushey) [reply src/node/desktop/CMakeLists.txt:219]: Makes sense! Thanks; let's keep it.

**Review by sharon-wang — APPROVED** [review]
LGTM too!

**Pull request merged**